### ISSUE TITLE
Improve diagnostic for missing `)` on function call

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#535]
quick-lint-js reports confusing diagnostics for the following code:
```javascript
  console.log("wrapNodes", wrapperElement, firstChildNode, lastChildNode
  lastChildNode.parentNode.insertBefore(
    wrapperElement,
    lastChildNode.nextSibling
  );
```
qljs should report something like "missing `)` in function call", pointing to the end of the `console.log` line.

**Comment by ahmafi** [comment]
I was reading the related codes for this issue and realized that this comes from this `parse_expression` call.

https://github.com/quick-lint/quick-lint-js/blob/d5a3bf8f605358b212ba27522f0c73b82ccf4245/src/parse.cpp#L1407-L1408

I could solve the issue with only changing `.trailing_identifiers = true` to `.trailing_identifiers = false`. One of the tests can't pass after this change. The test is something like the second line of these screenshots.

`.trailing_identifiers = true`:
![image](https://user-images.githubusercontent.com/12560109/144712933-196bc4cf-9016-4141-82e1-df91052884f6.png)

`.trailing_identifiers = false`:
![image](https://user-images.githubusercontent.com/12560109/144712895-05ab6991-ad4e-418a-a07f-440878583dc4.png)

As you can see there is no such a problem with `token_type::string`.

I think this can be the expected behaviour but I'm not sure. Also, V8 treats this issue like this:

![image](https://user-images.githubusercontent.com/12560109/144713538-e5689855-7b5b-4e6b-b4b3-e2aa8a7395e6.png)

@strager If you think this solves the issue I can make a PR.

**Comment by strager** [comment]
For both of your examples (`console.log(a b)` and `console.log("a" "b")`), the user is missing a comma. I think reporting E0147 is the right thing to do.

This is a hard problem, because we don't know *locally* if the problem is a missing `,` or a missing `)`.

```javascript
// case 1
console.log("a", b, c, d  // missing ')'
hello(world);

// case 2
console.log("a", b, c, d  // missing ','
  hello(world));

// case 3
console.log(
 "a",
  b  // missing ','
  hello(world),
);
```

Some ideas:
* Speculatively parse, assuming a `,`. If we reach the end of a expression but don't see `)`, error with 'missing `)` for function call'. Sounds complicated.
* Report 'missing `)` for function call' if there's a newline, or report 'missing `,` for function call` if there is no newline. This would make cases 1 and 2 above work, but not case 3. This might be the most pragmatic solution for now.
